model/src/main/java/org/projectnessie/api/params/ReferencesParams.java

```diff
@@ -29,12 +32,24 @@
 
   public ReferencesParams() {}
 
-  private ReferencesParams(Integer maxRecords, String pageToken) {
+  @Parameter(
+      description =
+          "If set to true, will fetch additional metadata for branches, such as number of commits ahead/behind or the common ancestor in relation to the default branch, and the commit metadata for the HEAD commit. "
+              + "Note that computing & fetching additional metadata might be computationally expensive on the server-side, so this flag should be used with care.")
+  @QueryParam("fetchMetadata")
+  private boolean fetchAdditionalMetadata;
```

Comment:
We also need another parameter to specify the reference-type to fetch (only branches, only tags, branches + tags).
Reason is that computing the common-ancestor + commits-behind/ahead is expensive.
Similar to https://github.com/snazy/nessie/branches vs https://github.com/snazy/nessie/tags

Reply:
let's do this in a follow-up PR as otherwise that would pollute this one

Reply:
It's just a single parameter, but it requires additional testing and it's not exactly in the scope of "fetch me all metadata". Therefore I think it's cleaner to have this as a separate PR